sale_order_deliverable_rate/models/sale_order.py

```diff
@@ -0,0 +1,48 @@
+from odoo import api, fields, models
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    deliverable_rate = fields.Float(
+        compute="_compute_deliverable_rate",
+        search="_search_deliverable_rate",
+        digits="Sale order deliverable rate",
+        store=False,
+    )
+
+    no_order_line_need_delivery = fields.Float(
+        compute="_compute_order_line_need_delivery", store=False
+    )
+
+    def order_line_need_delivery(self):
+        self.ensure_one()
+        return self.order_line.filtered(
+            lambda line: line.product_id.type == "product" and line.qty_to_ship > 0
+        )
+
+    @api.depends("order_line")
+    def _compute_order_line_need_delivery(self):
+        for rec in self:
+            rec.no_order_line_need_delivery = len(rec.order_line_need_delivery())
+
+    def _compute_deliverable_rate(self):
+        for rec in self:
+
+            if rec.no_order_line_need_delivery > 0:
+                rec.deliverable_rate = (
+                    sum(rec.order_line_need_delivery().mapped("deliverable_rate"))
+                    / rec.no_order_line_need_delivery
+                )
+            else:
+                # if everything is delivered deliverable rate => 0
+                rec.deliverable_rate = 0
+
+    def _search_deliverable_rate(self, operator, value):
+        lines = self.env["sale.order.line"].search([("qty_to_ship", ">=", 0)])
+
+        orders = lines.mapped("order_id")
```

Comment:
use a read-group for better performances:

order_data = self.env["sale.order.line"].read_group([("qty_to_ship", ">=", 0)], ["qty_to_ship"], ["order_id"])
order_ids = [data['order_id'][0] for data in order_data]
orders = self.browse(order_ids)